llm_judge/llm_judge.py

```diff
@@ -0,0 +1,1141 @@
+        :param tokenizer_judge_config: the config of the tokenizer to use for grading
+        :param model_bench_mark: the model to use for bench marking
+        :param model_bench_mark_config: the config of the model to use for bench marking
+        :param tokenizer_bench_mark_config: the config of the tokenizer to use for bench marking
+        :param model_bench_mark_infer_config: the config of the model to use for inference
+        :param prompt_template: the template of the prompt to use
+        :param prompt_config: the config of the prompt to use
+        """
+        super().__init__(
+            name,
+            model_judge,
+            model_judge_config,
+            model_bench_mark,
+            model_bench_mark_config,
+            model_bench_mark_infer_config,
+            tokenizer_bench_mark_config,
+            prompt_config,
+            prompt_template,
+            tokenizer_judge_config,
+            model_judge_infer_config,
+        )
+
+    def _compute_over_one_data(self, question, response, reference) -> Dict[str, Any]:
+        """
+        Compute the metrics over one data point
+        :param kwargs: the data to compute the metrics over
+        :returns: the metrics score and the explanation
+        """
+        self.prompt_config["reference"] = reference
+        res_dic = super()._compute_over_one_data(question, response)
+        return res_dic
+
+    @_open_mpi_handler(worker_inputs="sample_df")
+    def _compute_over_data(
+        self, sample_df: pd.DataFrame, train_df: pd.DataFrame = None
+    ) -> pd.DataFrame:
+        """
+        Compute the metrics over a dataset
+        :param sample_df: the data to compute the metrics over
+        :returns: the metrics score and the explanation
+        """
+        self._prepare_judge()
+        self._prepare_bench_mark_model()
+        res_df = pd.DataFrame(
+            columns=[
+                "question",
+                "answerA",
+                "answerB",
+                "reference",
+                "score_of_assistant_a",
+                "explanation_of_assistant_a",
+                "score_of_assistant_b",
+                "explanation_of_assistant_b",
+            ]
+        )
+
+        for i in range(len(sample_df)):
+            res_dic = self._compute_over_one_data(
+                sample_df.loc[i, "question"],
+                sample_df.loc[i, "answer"],
+                sample_df.loc[i, "reference"],
+            )
+            res_df.loc[i] = [
+                sample_df.loc[i, "question"],
+                sample_df.loc[i, "answer"],
+                sample_df.loc[i, "reference"],
+                res_dic["answerB"],
+                res_dic["score_of_assistant_a"],
+                res_dic["explanation_of_assistant_a"],
+                res_dic["score_of_assistant_b"],
+                res_dic["explanation_of_assistant_b"],
+            ]
+
+        return res_df
+
+
+class OPENAIJudgeSingleGrading(LLMJudgeSingleGrading):
```

Comment:
Do you need to do everything again for OpenAi? no way to integrate it better and re-use some code?

Reply:
Looking into it .